python-mini-projects/projects/Create_calculator_app/calculator.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+from tkinter import Tk, END, Entry, N, E, S, W, Button
+from tkinter import font
+from tkinter import Label
+from functools import partial
+
+
+def get_input(entry, argu):
+    entry.insert(END, argu)
+
+
+def backspace(entry):
+    input_len = len(entry.get())
+    entry.delete(input_len - 1)
+
+
+def clear(entry):
+    entry.delete(0, END)
+
+
+def calc(entry):
+    input_info = entry.get()
+    try:
+        output = str(eval(input_info.strip()))
+    except ZeroDivisionError:
+        popupmsg()
+        output = ""
+    clear(entry)
+    entry.insert(END, output)
+
+
+def popupmsg():
+    popup = Tk()
+    popup.resizable(0, 0)
+    popup.geometry("120x100")
+    popup.title("Alert")
+    label = Label(popup, text="Cannot divide by 0 ! \n Enter valid values")
+    label.pack(side="top", fill="x", pady=10)
+    B1 = Button(popup, text="Okay", bg="#DDDDDD", command=popup.destroy)
+    B1.pack()
+
+
+def cal():
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 61 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 61-99 into a separate function.